core/flow/warm_up.go

```diff
@@ -0,0 +1,127 @@
+package flow
+
+import (
+	"math"
+	"sync/atomic"
+
+	"github.com/alibaba/sentinel-golang/util"
+
+	"github.com/alibaba/sentinel-golang/core/base"
+)
+
+type WarmUpTrafficShapingCalculator struct {
+	threshold float64
+}
+
+func NewWarmUpTrafficShapingCalculator(threshold float64) *WarmUpTrafficShapingCalculator {
+	return &WarmUpTrafficShapingCalculator{threshold: threshold}
+}
+
+func (d *WarmUpTrafficShapingCalculator) CalculateAllowedTokens(base.StatNode, uint32, int32) float64 {
+	return d.threshold
+}
+
+type WarmUpTrafficShapingChecker struct {
+	metricType        MetricType
+	warmUpPeriodInSec uint32
+	coldFactor        uint32
+	warningToken      uint64
+	maxToken          uint64
+	slope             float64
+	storedTokens      *uint64
+	lastFilledTime    *uint64
+	count             float64
+}
+
+func NewWarmUpTrafficShapingChecker(metricType MetricType, warmUpPeriodInSec, coldFactor uint32, count float64) *WarmUpTrafficShapingChecker {
```

Comment:
Actually only `WarmUpTrafficShapingCalculator` is needed, which is designed for calculating the threshold during the warm-up period (or equal to the original threshold after the warm-up period has elapsed). Then we could just combine the calculator with the universal `DefaultTrafficShapingChecker`.